src/main/java/org/primefaces/component/datepicker/DatePickerBase.java

```diff
@@ -386,4 +387,35 @@ public void setOnYearChange(String onYearChange) {
     public boolean hasTime() {
         return this.isShowTime() || this.isTimeOnly() || super.hasTime();
     }
-}
\ No newline at end of file
+
+    @Override
+    public String calculatePattern() {
```

Comment:
Please add a few tests for this method. Eg userdefinied pattern where we need to remove time. TimeOnlyPattern with 12 and 24 hours.